akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -105,25 +105,52 @@ trait SupervisorStrategyLowPriorityImplicits { this: SupervisorStrategy.type =>
   // the above would clash with seqThrowable2Decider for empty lists
 }
 
+@nowarn("msg=deprecated")
 object SupervisorStrategy extends SupervisorStrategyLowPriorityImplicits {
   sealed trait Directive
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] sealed trait DirectiveWithLevel {
+    def logLevel: LogLevel
+  }
+
   /**
    * Resumes message processing for the failed Actor
    */
+  @Deprecated
+  @deprecated("use resume method with log level instead", "2.6.15")
   case object Resume extends Directive
```

Comment:
I think each of these should be kept, and not deprecated, and have the default loglevel. So something like `case object Resume extends Resume(LogLevel.Info)`, I think that can also keep bincomp with the Java getters.

Reply:
My intention was to get rid of the objects and just keep the methods with and without the log level argument (similar to to `OverflowStrategy` since calling code does not have to match on those Directives itself.

But extending the object from case class would also work for me.

Reply:
It's good because it will keep the current shared instances by default instead of creating new instances unless needed.